ui/src/main/java/com/wireguard/android/fragment/TunnelListFragment.kt

```diff
@@ -52,7 +54,20 @@ class TunnelListFragment : BaseFragment() {
         val activity = activity ?: return@registerForActivityResult
         val contentResolver = activity.contentResolver ?: return@registerForActivityResult
         activity.lifecycleScope.launch {
-            TunnelImporter.importTunnel(contentResolver, data) { showSnackbar(it) }
+            if (contentResolver.getType(data)?.startsWith("image/") == true) {
+                val qrCodeFromFileScanner = QrCodeFromFileScanner(contentResolver, QRCodeReader())
+                val qrResult = qrCodeFromFileScanner.scan(data)
+                qrResult.onSuccess {
+                    TunnelImporter.importTunnel(parentFragmentManager, it.text) { showSnackbar(it) }
+                }.onFailure { throwable ->
+                    val error = ErrorMessages[throwable]
+                    val message = requireContext().getString(R.string.import_error, error)
```

Comment:
I assume this will go away once we move the above logic to `TunnelImporter`? We also don't use Kotlin's `Result` type anywhere in this codebase and I don't really intend to start now, so I'd prefer if it was eliminated as well, there's existing machinery in `TunnelImporter` to handle errors that we can reuse.